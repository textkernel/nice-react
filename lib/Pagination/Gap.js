import React, { Component } from 'react';

export default class Gap extends Component {
    render() {
        return (
            <li className="disabled">
                <span>&hellip;</span>
            </li>
        );
    }
}
