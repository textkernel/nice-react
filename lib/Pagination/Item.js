import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Item extends Component {
    handleClick(e) {
        const { onClick } = this.props;
        e.preventDefault();
        onClick();
    }

    render() {
        const { current, children, disabled } = this.props;

        const classes = [];
        if (current) {
            classes.push('current');
        }
        if (disabled) {
            classes.push('disabled');
        }

        const NodeType = disabled ? 'span' : 'a';

        return (
            <li
                className={ classes }
            >
                <NodeType
                    onClick={ !disabled ? e => { this.handleClick(e); } : null }
                    href={ !disabled ? '#' : '' }
                >
                    { children }
                </NodeType>
            </li>
        );
    }
}

Item.propTypes = {
    current: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
        PropTypes.number
    ]),
    onClick: PropTypes.func,
    disabled: PropTypes.bool
};

Item.defaultProps = {
    current: false,
    children: '',
    onClick: null,
    disabled: false
};

export default Item;
