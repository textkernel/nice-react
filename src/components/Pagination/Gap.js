import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';


const Gap = ({ className, style }) => (
    <li className={classnames('disabled', className)} style={style}>
        <span>&hellip;</span>
    </li>
);

Gap.propTypes = {
    className: PropTypes.string,
    style: PropTypes.object,
};

Gap.defaultProps = {
    className: null,
    style: {}
};

export default Gap;
