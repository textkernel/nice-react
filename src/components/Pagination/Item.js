import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

class Item extends Component {
    handleClick(e) {
        const { onClick } = this.props;
        e.preventDefault();
        onClick();
    }

    render() {
        const { current, children, disabled, className, style } = this.props;

        const classes = [];
        if (current) {
            classes.push('current');
        }
        if (disabled) {
            classes.push('disabled');
        }

        const NodeType = disabled ? 'span' : 'a';

        return (
            <li
                className={classnames(classes.join(' '), className)}
                style={style}
            >
                <NodeType
                    onClick={ !disabled ? e => { this.handleClick(e); } : null }
                    href={ !disabled ? '#' : '' }
                >
                    { children }
                </NodeType>
            </li>
        );
    }
}

Item.propTypes = {
    className: PropTypes.string,
    current: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
        PropTypes.number
    ]),
    style: PropTypes.object,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
};

Item.defaultProps = {
    className: null,
    style: {},
    current: false,
    children: '',
    onClick: null,
    disabled: false
};

export default Item;
