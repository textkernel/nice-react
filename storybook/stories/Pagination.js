import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobsOptions, withKnobs, boolean, number, select } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import props from '../../src/props';

import { Pagination } from 'nice-react';

const stories = storiesOf('Pagination', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('Default', () => (
    <Pagination>
        <Pagination.Item disabled>&laquo; Previous</Pagination.Item>
        <Pagination.Item current onClick={ () => { window.alert(1); }}>1</Pagination.Item>
        <Pagination.Item onClick={ () => { window.alert(2); }}>2</Pagination.Item>
        <Pagination.Item onClick={ () => { window.alert(3); }}>3</Pagination.Item>
        <Pagination.Item onClick={ () => { window.alert(4); }}>4</Pagination.Item>
        <Pagination.Item onClick={ () => { window.alert(5); }}>5</Pagination.Item>
        <Pagination.Item>Next &raquo;</Pagination.Item>
    </Pagination>
));

stories.add('Bordered', () => (
    <Pagination bordered>
        <Pagination.Item disabled>&laquo; Previous</Pagination.Item>
        <Pagination.Item current>1</Pagination.Item>
        <Pagination.Item>2</Pagination.Item>
        <Pagination.Item>3</Pagination.Item>
        <Pagination.Item>4</Pagination.Item>
        <Pagination.Item>5</Pagination.Item>
        <Pagination.Item>Next &raquo;</Pagination.Item>
    </Pagination>
));

stories.add('With gap', () => (
    <Pagination bordered>
        <Pagination.Item disabled>&laquo; Previous</Pagination.Item>
        <Pagination.Item current>1</Pagination.Item>
        <Pagination.Item>2</Pagination.Item>
        <Pagination.Item>3</Pagination.Item>
        <Pagination.Gap />
        <Pagination.Item>9</Pagination.Item>
        <Pagination.Item>10</Pagination.Item>
        <Pagination.Item>Next &raquo;</Pagination.Item>
    </Pagination>
));

stories.add('Small', () => (
    <Pagination size="small">
        <Pagination.Item disabled>&laquo; Previous</Pagination.Item>
        <Pagination.Item current>1</Pagination.Item>
        <Pagination.Item>2</Pagination.Item>
        <Pagination.Item>3</Pagination.Item>
        <Pagination.Item>4</Pagination.Item>
        <Pagination.Item>5</Pagination.Item>
        <Pagination.Item>Next &raquo;</Pagination.Item>
    </Pagination>
));

stories.add('Large', () => (
    <Pagination size="large">
        <Pagination.Item disabled>&laquo; Previous</Pagination.Item>
        <Pagination.Item current>1</Pagination.Item>
        <Pagination.Item>2</Pagination.Item>
        <Pagination.Item>3</Pagination.Item>
        <Pagination.Item>4</Pagination.Item>
        <Pagination.Item>5</Pagination.Item>
        <Pagination.Item>Next &raquo;</Pagination.Item>
    </Pagination>
));

stories.add('Custom properties', () => {
    const sizes = select('Size', props.paginationSizes);
    const bordered = boolean('Bordered', false);
    const prevDisabled = !boolean('Previous enabled', false);
    const nextDisabled = !boolean('Next enabled', true);
    const pages = number('Number of pages', 5);
    const currentPage = number('Current page', 1);

    return <Pagination size={ sizes } bordered={ bordered }>
        <Pagination.Item disabled={ prevDisabled }>&laquo; Previous</Pagination.Item>
        {
            Array(pages).fill(0).map((v, k) => (
                <Pagination.Item current={ currentPage && k + 1 === currentPage }>
                    { k + 1 }
                </Pagination.Item>
            ))
        }
        <Pagination.Item disabled={ nextDisabled }>Next &raquo;</Pagination.Item>
    </Pagination>;
});
