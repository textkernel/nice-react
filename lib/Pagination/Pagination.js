import React, { Component } from 'react';
import PropTypes from 'prop-types';
import niceProps from '../props';
import Item from './Item';
import Gap from './Gap';

export default class Pagination extends Component {
    static propTypes = {
        size: PropTypes.oneOf(niceProps.paginationSizes),
        bordered: PropTypes.bool,
        children: PropTypes.node
    }

    static defaultProps = {
        size: null,
        bordered: false,
        children: null
    }

    render() {
        const { size, bordered, children } = this.props;

        let classes = ['nice-pagination'];
        if (size) {
            classes.push(size);
        }
        if (bordered) {
            classes.push('bordered');
        }
        classes = classes.join(' ');

        return (
            <nav>
                <ul className={ classes } role="navigation">
                    { children }
                </ul>
            </nav>
        );
    }
}

Pagination.Item = Item;
Pagination.Gap = Gap;