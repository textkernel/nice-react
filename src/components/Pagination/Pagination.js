import React from 'react';
import PropTypes from 'prop-types';
import niceProps from '../../props';
import Item from './Item';
import Gap from './Gap';
import './Pagination.scss';

const Pagination = ({ size, bordered, children, className, style }) => {
    let classes = ['nice-pagination'];
    if (size) {
        classes.push(size);
    }
    if (bordered) {
        classes.push('bordered');
    }
    classes = classes.join(' ');

    return (
        <nav className={className} style={style}>
            <ul
                className={ classes }
                role="navigation"
            >
                { children }
            </ul>
        </nav>
    );
};

Pagination.propTypes = {
    className: PropTypes.string,
    style: PropTypes.object,
    size: PropTypes.oneOf(niceProps.paginationSizes),
    bordered: PropTypes.bool,
    children: PropTypes.node
};

Pagination.defaultProps = {
    className: null,
    style: {},
    size: null,
    bordered: false,
    children: null
};

Pagination.Item = Item;
Pagination.Gap = Gap;

export default Pagination;
