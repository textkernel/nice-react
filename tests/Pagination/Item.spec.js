import React from 'react';
import Pagination from '../../lib/Pagination';

describe('<Pagination.Item />', () => {
    it('should render current Item', () => {
        const onClickHandler = jest.fn();
        const wrapper = shallow(<Pagination.Item onClick={ onClickHandler } current />);
        expect(wrapper).toMatchSnapshot();

        wrapper.find('li.current a').simulate('click', { preventDefault() {} });
        expect(onClickHandler).toHaveBeenCalled();
    });

    it('should render disabled Item', () => {
        const onClickHandler = jest.fn();
        const wrapper = shallow(<Pagination.Item onClick={ onClickHandler } disabled />);
        expect(wrapper).toMatchSnapshot();

        wrapper.find('li.disabled span').simulate('click', { preventDefault() {} });
        expect(onClickHandler).not.toHaveBeenCalled();
    });
});
