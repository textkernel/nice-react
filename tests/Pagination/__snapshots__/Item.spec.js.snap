// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination.Item /> should render current Item 1`] = `
<li
  className={
    Array [
      "current",
    ]
  }
>
  <a
    href="#"
    onClick={[Function]}
  />
</li>
`;

exports[`<Pagination.Item /> should render disabled Item 1`] = `
<li
  className={
    Array [
      "disabled",
    ]
  }
>
  <span
    href=""
    onClick={null}
  />
</li>
`;
